RecoTracker/LSTCore/standalone/code/core/write_lst_ntuple.cc

```diff
@@ -945,14 +945,14 @@ std::tuple<float, float, float, std::vector<unsigned int>, std::vector<unsigned
 //________________________________________________________________________________________________________________________________
 void printHitMultiplicities(Event* event) {
   Modules const* modules = event->getModules().data();
-  ObjectRanges const* ranges = event->getRanges().data();
+  auto ranges = event->getRanges<ObjectRangesSoA>();
 
   int nHits = 0;
   for (unsigned int idx = 0; idx <= *(modules->nLowerModules);
        idx++)  // "<=" because cheating to include pixel track candidate lower module
   {
-    nHits += ranges->hitRanges[4 * idx + 1] - ranges->hitRanges[4 * idx] + 1;
-    nHits += ranges->hitRanges[4 * idx + 3] - ranges->hitRanges[4 * idx + 2] + 1;
+    nHits += ranges.hitRanges()[2 * idx][1] - ranges.hitRanges()[2 * idx][0] + 1;
+    nHits += ranges.hitRanges()[2 * idx + 1][1] - ranges.hitRanges()[2 * idx][0] + 1;
   }
```

Comment:
@slava77 this loop looks weird. Would it be okay to change it to

```cpp
for (unsigned int idx = 0; idx <= *(modules->nModules); idx++) {
  nHits += ranges.hitRanges()[idx][1] - ranges.hitRanges()[idx][0] + 1;
}
```
or was there a reason for having it like that?

Reply:
IIRC, the hitRanges over all modules.
The loop is over `nLowerModules`.
beyond the ntuple here, I can see some limited benefit from splitting hits to lower and upper module hits.